docs/reference-guides/block-api/block-registration.md

```diff
@@ -273,6 +275,31 @@ The `ancestor` property makes a block available inside the specified block types
 ancestor: [ 'core/columns' ],
 ```
 
+#### Block Hooks (optional)
+
+-   **Type:** `Object`
+-   **Since**: `WordPress 6.4.0`
+
+Block Hooks is the API that allows a block to hook into the rendering of another block or multiple blocks. It will enable a block to render its content before or after another block. Alternatively, it's possible to hook into the rendering of a parent block and prepend or append the block to the list of inner blocks. The key is the name of the block (`string`) to hook into, and the value is the position to hook into (`string`). Allowed target values are:
+
+-   `before` – inject before the target block.
+-   `after` - inject after the target block.
+-   `firstChild` - inject before the first inner block of the target container block.
+-   `lastChild` - inject after the last inner block of the target container block.
+
+```js
+{
+	blockHooks: {
+		'core/verse': 'before'
+		'core/spacer': 'after',
+		'core/column': 'firstChild',
+		'core/group': 'lastChild',
+	}
+}
+```
+
+It’s crucial to emphasize that the Block Hooks feature is designed to work only with static templates, template parts, and patterns loaded from HTML files shipped with the theme. It also integrates with patterns registered on the server by the site. The modification will never apply to the post content or patterns crafted by the user, nor theme templates and template parts modified by the user.
```

Comment:
Surfacing the comment from @fabiankaegy added on the commit:

https://github.com/WordPress/gutenberg/commit/7c0c3a0e815a0853af7db71e0745e057f8e882dc#r127115387

Reply:
👍 We could consider adding a note on why we're making that tradeoff (of not auto-inserting into templates with user modifications) (as explained [here](https://core.trac.wordpress.org/ticket/59313)) -- but maybe that's too technical for this more hand-on kind of API documentation.

Reply:
@colorful-tones plan to open a follow-up PR that will include further improvements to the docs as outlined in https://github.com/WordPress/gutenberg/commit/7c0c3a0e815a0853af7db71e0745e057f8e882dc#r127154283. Let's discuss that part alongside these changes.